### ISSUE TITLE
Create all parent directories

### ISSUE THREAD

**Issue by chshersh** [kowainik/hash-store#9]
`hashStore` function fails if executed with arguments like this:

```haskell
hashStore "dir" action ("src/Some/Module/Name.hs", ...)
```

I guess It should create all directories for file name.

**Comment by chshersh** [comment]
Creating directories is tricky... For example, which file should I create for `foo/bar`?

1. `hash-foo/bar`
2. `foo/hash-bar`

If second one, then property in documentation becomes more verbose...

**Comment by int-index** [comment]
I don't think that the store must allow any file hierarchy. If that was the goal, then the input should be `Tree (String, ByteString)` instead, it's a completely different model.
